### PR TITLE
Public API POST sign up route and controller

### LINKED ISSUE #19: Public API POST sign up route and controller
- `/signUp` route should receive username, password, e-mail, first & last name and phone number
- validate it using Express-Validator 
- save data in the database

### PR DESCRIPTION
**Pull request by JanaAbuHaltam**
closes #19

### CODE REVIEW

File: package.json
Comment (belalninja) [inline package.json:1]: we should install dependencies corresponding to the backend in path `server/package.json`, this global `package.json` file is just for project setup

File: server/server/Routes/signup.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+const router = express.Router();
+const { body } = require("express-validator");
+const userController = require("../controllers/userController");
+
+// Define the sign-up route
+router.post(
+  "/signup",
+
+  // Validation rules
+  body("username").isLength({ min: 5 }),
```
Comment (belalninja) [inline server/server/Routes/signup.js:11]: i think we should include a custom message with each validation rule to be sent in response, like:

```suggestion
  body("username", "Username should be at least 5 characters").isLength({ min: 5 }),
```

File: server/server/Routes/signup.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+  body("password").isLength({ min: 5 }),
+  body("email").isEmail(),
+  body("firstName").notEmpty(),
+  body("lastName").notEmpty(),
+  body("phoneNumber").isMobilePhone(),
```
Comment (belalninja) [inline server/server/Routes/signup.js:16]: i think we should include a custom message with each validation rule to be sent in response, feel free to accept these changes:

```suggestion
  body("username", "Username should be at least 5 characters").isLength({ min: 5 }),
  body("password", "Password should be at least 5 characters").isLength({ min: 5 }),
  body("email", "Email should be in 'example@email.com' format").isEmail(),
  body("firstName", "First Name should not be empty").notEmpty(),
  body("lastName", "Last Name should not be empty").notEmpty(),
  body("phoneNumber", "Phone Number should be a Jordanian mobile number").isMobilePhone('ar-JO'),
```

File: server/server/Controllers/userController.js [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    lastName: req.body.lastName,
+    phoneNumber: req.body.phoneNumber,
+  });
+
+  try {
+    // Save the user to the database
+    const savedUser = await user.save();
+    res
+      .status(201)
+      .json({ message: "User created successfully", user: savedUser });
```
Comment (belalninja) [inline server/server/Controllers/userController.js:38]: returning the `savedUser` will have property `password_hash`, i don't think it's a vulnerability, just a best practice; feel free to pass it or not

**Review by belalninja — CHANGES_REQUESTED** [review]
brilliant @JanaAbuHaltam ; just some changes requested

File: server/server/Controllers/userController.js (hunk 3 above)
Comment (JanaAbuHaltam) [reply server/server/Controllers/userController.js:38]: Can you check my edits and provide me with any possible changes? its more than welcome.
I just keep the `savedUser` part with `password_hash` untill i do research

**Review by belalninja — APPROVED** [review]
good job 👍

**Review by belalninja — APPROVED** [review]

**Pull request merged**